generator/src/main/java/com/algorand/sdkutils/generators/Generator.java

```diff
@@ -215,57 +205,65 @@ static TypeDef getType(
+                    type = prop.get("items").get("type").asText(); 
+                    return getBase64EncodedArray(propName, goName, type, imports, forModel, desc, isRequired(prop));
                 } else {
-                    return getBase64Encoded(propName, imports, forModel);
+                    return getBase64Encoded(propName, goName, type, imports, forModel, desc, isRequired(prop));
                 }
             case "AccountID":
                 break;
             case "BlockCertificate":
             case "BlockHeader":
                 addImport(imports, "java.util.HashMap");
-                return new TypeDef("HashMap<String,Object>");
+                return new TypeDef("HashMap<String,Object>", type, propName, goName, desc, isRequired(prop));
             }
         }
         switch (type) {
         case "integer":
-            return asObject ? new TypeDef("Long") : new TypeDef("long");
+            String longName = asObject ? "Long" : "long"; 
+            return new TypeDef(longName, type, propName, goName, desc, isRequired(prop));
         case "object":
         case "string":
-            return new TypeDef("String");
+            return new TypeDef("String", type, propName, goName, desc, isRequired(prop));
         case "boolean":
-            return asObject ? new TypeDef("Boolean") : new TypeDef("boolean");
+            String boolName = asObject ? "Boolean" : "boolean"; 
+            return new TypeDef(boolName, type, propName, goName, desc, isRequired(prop));
         case "array":
             JsonNode arrayTypeNode = prop.get("items");
             TypeDef typeName = getType(arrayTypeNode, asObject, imports, propName, forModel);
-            return new TypeDef("List<" + typeName.typeName + ">", typeName.def, "list");
+            return new TypeDef("List<" + typeName.javaTypeName + ">", typeName.rawTypeName,
+                    typeName.def, type, propName, goName, desc, isRequired(prop));
         default:
             throw new RuntimeException("Unrecognized type: " + type);
         }
     }
 
+    // getTypeFormat returns the additional type formatting information
+    // There could be multiple such tags in the spec file. This method knows which 
+    // one is relevant here. 
     public static String getTypeFormat(JsonNode typeNode, String propName) {
         String format = typeNode.get("x-algorand-format") != null ? typeNode.get("x-algorand-format").asText() : "";
         String type = typeNode.get("type").asText();
         format = typeNode.get("format") != null && format.isEmpty() ? typeNode.get("format").asText() : format;
-        format = typeNode.get("x-go-name") != null && format.isEmpty() ? typeNode.get("x-go-name").asText() : format;
-        if ((propName.equals("address") || propName.contentEquals("accountId")) &&
+        if ((propName.equals("address") || 
+                propName.contentEquals("account-id") || 
+                propName.contentEquals("AccountID")) &&
```

Comment:
Why are these fields hard coded? should we update the spec?

Reply:
Definitely helps to update the specs. 
This is around updating the specs.